pina/geometry/simplex.py

```diff
@@ -0,0 +1,218 @@
+import torch
+
+from pina.geometry import CartesianDomain
+from .location import Location
+from pina import LabelTensor
+from ..utils import check_consistency
+
+
+class SimplexDomain(Location):
+    """PINA implementation of a Simplex."""
+
+    def __init__(self, simplex_dict, labels, sample_surface=False):
+        """
+        :param simplex_dict: A dictionary with dict-key a string representing
+            the input variables for the pinn, and dict-value a list
+            representing vertices of the simplex.
+        :type simplex_dict: dict
+        :param labels: A list of labels for vertex components
+        :type labels: list[str]
+        :param sample_surface: A variable for choosing sample strategies. If
+            `sample_surface=True` only samples on the Simplex surface
+            frontier are taken. If `sample_surface=False`, no such criteria
+            is followed.
+        :type sample_surface: bool
+
+        .. warning::
+            Sampling for dimensions greater or equal to 10 could result
+            in a shrinking of the simplex, which degrades the quality
+            of the samples. For dimensions higher than 10, other algorithms
+            for sampling should be used.
+
+        :Example:
+
+            >>> spatial_domain = SimplexDomain({'vertex1': [0, 0], 'vertex2': [1, 1], 'vertex3': [0, 2]}, ['x', 'y'])
+        """
+
+        # check consistency of labels
+        if not isinstance(labels, list):
+            raise ValueError(f"{type(labels).__name__} must be {list}.")
+        check_consistency(labels, str)
+        self._coordinate_labels = labels
+
+        # check consistency of sample_surface
+        check_consistency(sample_surface, bool)
+        self._sample_surface = sample_surface
+
+        # check consistency of simplex_dict
+        check_consistency(simplex_dict, dict)
+        for vertex in simplex_dict.values():
+            if not isinstance(vertex, list):
+                raise ValueError(f"{type(vertex).__name__} must be {list}.")
+
+        # vertices, vectors, dimension
+        self._vertices = simplex_dict
+        self._vert_list = torch.tensor(list(simplex_dict.values())).T
+        self._vectors = (
+            self._vertices_list[:, :-1] - self._vertices_list[:, None, -1]
+        ).type(torch.FloatTensor)
+
+        # build cartesian_bound
+        self._cartesian_bound = self._build_cartesian(
+            list(simplex_dict.values()), labels
+        )
+
+    @property
+    def variables(self):
+        """
+        Coordinate labels of simplex.
+
+        :return: Coordinate labels
+        :rtype: list[str]
+        """
+
+        return self._coordinate_labels
+
+    @property
+    def vertices(self):
+        """
+        Vertices of simplex.
+
+        :return: Vectors
+        :rtype: list[list]
+        """
+
+        return self._vertices
+
+    @property
```

Comment:
I don't think the user should access this. Please remove it.

Reply:
Removed this property